README.md

```diff
@@ -6,35 +6,248 @@
+)
+
+## Write to blob storage
+az_write_blob(
+  my_data,
+  container_name = "my_container",
+  path = "path/to/blob.ext"
+)
+```
+
+# Getting started
 
 ## Installation
 
 You can install the development version of cfazuR from
 [GitHub](https://github.com/) with:
 
 ``` r
-# install.packages("pak")
 pak::pak("CDCgov/cfazuR")
 ```
 
-## Example
+## Required setup after installation
+
+To connect to Azure storage resources with cfazuR, you will need to be
+able to authenticate (i.e. log in) to Azure from R. You also need to
+look up certain Azure credentials (i.e. account information) and store
+them in a location that is secure, but accessible to R.
+
+Follow these the steps carefully the first time you set up the package,
+and refer back to this page if you ever need to update your credentials.
+
+### 1. Learn how to keep your credentials secure
+
+**Big picture: Treat your credentials like passwords.** Do not commit
+them to GitHub, post them on the internet, send them in unencrypted
+emails, etc. This is especially important for service principals, which
+are essentially passwords. But it’s a good idea to follow these security
+guidelines for all of your credentials, the same way you’re probably
+already used to doing for other kinds of usernames and account numbers.
+
+**1.1 [Install and active
+detect-secrets](https://potential-adventure-637em36.pages.github.io/docs/pre_commit.html#getting-started-installing-pre-commit)**
+as a pre-commit hook. This will check your code for secrets before you
+commit to GitHub and raise an error if it detects a problem.
+
+**1.2 Never hard-code your credentials**
+
+``` r
+## WHAT NOT TO DO:
+az_login(
+  tenant = "1234-abcd-0987-zyxw",
+  app = "my-actual-client-id",
+  passowrd = "my-actual-service-principal"
+)
+```
+
+If you accidentally push this code to GitHub, your secrets are now on
+the internet. Instead, write code that represents your credentials as
+variables wihtout ever showing the underlying values:
+
+``` r
+## DO THIS INSTEAD
+##  Load credentials stored in a separate, .json, .yaml, or .toml file
+cred_list <- read_json(my_credentials.json)
+## or
+## Store credentials as environment variables and load them at runtime
+cred_list <- cfazuR::get_env_creds() # Load env vars
+az_login(
+  tenant = cred_list$az_tenant_id,
+  app = cred_list$az_client_id,
+  passowrd = cred_list$az_service_principal
+)
+```
+
+### 2. Look up your Azure Credentials
+
+To log in using [client-based authentication]() you will need:
+
+1.  Client ID
+2.  Tenant ID
+3.  Service principal
+
+To find these, log in to the Azure portal (<https://portal.azure.com/>),
+then search for “app registrations” and click on the relevant icon. If
+an app registration exits, click on it to reveal a page showing the
+associated client ID, and tenant ID. From this page, click again below
+“Certificates and secrets” to reveal the service principal value.
+
+:info: If no service principal exists under “App Registrations”, contact
```

Comment:
```suggestion
> [!NOTE]
> If no service principal exists under “App Registrations”, contact
```
Use [github's](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) alert system